app/grandchallenge/challenges/admin.py

```diff
@@ -132,6 +138,111 @@ def send_status_update_email(self, request, queryset):
             )
 
 
+class OnTimeFilter(admin.SimpleListFilter):
+    title = _("on time")
+    parameter_name = "on_time"
+
+    def lookups(self, *_, **__):
+        return [
+            ("yes", "Yes"),
+            ("no", "No"),
+        ]
+
+    def queryset(self, request, queryset):
+        if self.value() == "yes":
+            queryset = queryset.filter(overdue=False)
+        elif self.value() == "no":
+            queryset = queryset.filter(overdue=True)
+        return queryset
+
+
+@admin.action(
+    description="Mark selected onboarding tasks complete",
+    permissions=("change",),
+)
+def mark_task_complete(modeladmin, request, queryset):
+    queryset.update(complete=True)
+
+    modeladmin.message_user(
+        request, f"{len(queryset)} Tasks marked as complete", messages.SUCCESS
+    )
+
+
+@admin.action(
+    description="Move selected task' deadlines by 1 week",
+    permissions=("change",),
+)
+def move_task_deadline_1_week(modeladmin, request, queryset):
+    queryset.update(deadline=F("deadline") + timedelta(weeks=1))
+
+    modeladmin.message_user(
+        request,
+        f"{len(queryset)} task deadlines moved 1 week",
+        messages.SUCCESS,
+    )
+
+
+@admin.action(
+    description="Move selected task' deadlines by 4 weeks",
+    permissions=("change",),
+)
+def move_task_deadline_4_weeks(modeladmin, request, queryset):
+    queryset.update(deadline=F("deadline") + timedelta(weeks=4))
+
+    modeladmin.message_user(
+        request,
+        f"{len(queryset)} task deadlines moved 4 weeks",
+        messages.SUCCESS,
+    )
+
+
+@admin.register(OnboardingTask)
+class OnboardingTaskAdmin(ModelAdmin):
+    ordering = (
+        "challenge",
+        "-deadline",
+    )
+    autocomplete_fields = ("challenge",)
+    list_display = (
+        "title",
+        "challenge",
+        "on_time",
+        "complete",
+        "deadline",
+        "responsible",
+    )
+    list_filter = (
+        OnTimeFilter,
+        "complete",
+        "challenge__short_name",
+    )
+    list_select_related = ("challenge",)
+    search_fields = ("title", "description")
+    actions = (
+        mark_task_complete,
+        move_task_deadline_1_week,
+        move_task_deadline_4_weeks,
+    )
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self._reference_time = now()
+
+    def get_queryset(self, request):
+        qs = super().get_queryset(request)
+        return qs.annotate(
```

Comment:
Use database-based annotation here for both speed and the option to filter things.